### DISCUSSION TITLE
A few things I'd like to try to code into this mod - if nobody thinks they're bad ideas.

### DISCUSSION THREAD

**Discussion by DoubleTheAndreno** [Monika-After-Story/MonikaModDev#8136] [QA - code and content creation (technical)]
Hello.  

I really like the idea of creating "virtual things that try to act like people" so this game really caught my interest and I feel like it might be fun to try to contribute to.  Before I go trying anything though, I wanted to ask some questions and propose some of the things I'd like to do so you all have a chance to shoot them down.

I know how to program (or I sure hope I do) well enough, so I should be able to realize most of what I suggest here if I stick with it. That said, I'm super new to this game and I've read through the code base to get a feel for how stuff works but I haven't had a chance to really deeply understand it yet.   

Forgive me if some of the stuff I mention here already exists.

And some of this may be super hard to accomplish. That's alright, and  I'm willing to try and fail to build it myself.  But have you already tried some of this and it just didn't work?  Are any of you opposed to these ideas based on your ideals on what is good game design?  Any reason you'd see a wrapped up well coded pull request with one of these things and reject it because it just doesn't suit you?

Without further ado, here's some stuff I'd like to try:

1. Some sort of internal state when it comes to Monika's current idle random requests.  I'd love to see Monika to go through something like "I want the player to interact with me so if the player has just been reading my comments I should ask them a question".  Or "I feel a need to contribute to the player's life so I reach out to offer them fun facts that might make them happy".  Say the player reaches out and says good morning to Monika, that might fill a meter of "I'm being paid attention to".

2. A log of events (like the audit file you've already got for affection, but beefed up) and their positivity.  

2.5.  Clustering the log of events by day of week and time so Monika can have an "expectation" that the player is going to interact with her negatively or positively at a given moment.  Say the player is in a habit of logging in and saying good morning.  The game picks up on that and if the player misses a good morning, Monika is actually able to comment on that and a little"disappointment" meter will fill up.  (Or the opposite of that - if there is a regular negative interaction that would create "fear" and Monika might prompt the player to avoid whatever it is they do at that time, maybe their computer is scheduled to reboot and they aren't saying bye) 

3. It looks like most all the game dialog is hard coded.  Could that dialog be moved out to a "random picker system" where anyone can just come in and dump an extra bit of text onto the end of a file to add alternative dialog options?

4. Some sort of live dialog-suggestion-approval system that ties up to a website instead of needing a github issue/pull request?  Say a user wants to suggest a new topic/dialog option, they pick a little prompt in game - pick when the dialog happens(maybe redirect them to a webpage) - and enter dialog.  You see this as the project maintainers and you can press a button to auto-build a new file with the added dialog and credits.  

5. The "Unused" dialog box option right now shows negative options.  I do not want to use them, but there's no clear way to get rid of them.  Some option of "that's horrible I don't want to say that" to hide them without saying them would be good.  

6. Monika can send out notifications, but there aren't many consequences to sitting and ignoring her.  I'd love the option to have her get increasingly frustrated if she says "hey talk to me" and I ignore her for 30 minutes.

Here's my impression of each:

1 and 2 are the big ones that I'd love to play with.

3 is less exciting, but I think important for a game like this one.  Not only for the ability to add more dialog, but so (insert X here who doesn't speak English) can write translations.

4 is a bit nuts/over the top, but I hate repeat dialog and the best way to not have repeat dialog without some stupid supernatural AI is to have ALL THE DIALOG.  The more you can encourage/make it easier for people to suggest new stuff, the better.  I could even build a little javascript based parser/simulator for all the pauses and delays and deletes so you could see what the dialog looks like live.

5 is a little quality of life thing I'd probably do to get started with a first pull request. 

6 would be annoying as heck and should be optional, but would feel realistic.  Probably second on the list of "easier changes to make to get my feet under me".

What are your thoughts here?  Is any of this feasible?

**Comment by Booplicate** [comment]
> 2. A log of events and their positivity.

What do you mean by that? If you want to just log event data, it's already being logged.

> 3. It looks like most all the game dialog is hard coded. Could that dialog be moved out to a "random picker system" where anyone can just come in and dump an extra bit of text onto the end of a file to add alternative dialog options?

You can't just "dump" it, you have to specify where and how the option would appear, which expression Monika would have. How moving it from the random selection system would help?

renpy has its own system to handle translations, we don't directly support them, but you can make one using renpy documentation.

> 4. Some sort of live dialog-suggestion-approval system that ties up to a website instead of needing a github issue/pull request? Say a user wants to suggest a new topic/dialog option, they pick a little prompt in game - pick when the dialog happens(maybe redirect them to a webpage) - and enter dialog. You see this as the project maintainers and you can press a button to auto-build a new file with the added dialog and credits.

New topics go through the process of reviewing and approving. That's to keep the quality level and standards. It's also a much more complex system. Start/end dates, conditionals, actions, rules, different databases and other props. Labels can have python code blocks, renpy lang statements (show/hide img/menu/ call/jump to a label), character expressions, etc. Much easier to work in an actual code editor than using some kind of "auto-topic-builder."

Regarding repeating topics, we're focusing on making each topic repeatable (to an extent), it's impossible to have unlimited number of conversations. There were a few people who wanted to implement some training models for MAS, but those never went anywhere (or at least we aren't aware of it).

> 5. The "Unused" dialog box option right now shows negative options.

#7912

**Comment by briaryoung** [comment]
For your response (that may have been deleted, I'm not seeing it here)- No. We don't intend to be rude. We like to keep our professionalism and sometimes that can lead to seeming unfriendly. We give our honest opinions about what may or may not go into the mod. We appreciate you wanting to contribute. We appreciate you bringing your ideas to us first- that way your code would not have gone to waste.

**Comment by briaryoung** [comment]
Also, No. 1 has not been rejected.
